source/non_matching/mesh_classifier.cc

```diff
@@ -68,9 +68,11 @@ namespace NonMatching
           std::minmax_element(local_levelset_values.begin(),
                               local_levelset_values.end());
 
-        if (*min_max_element.second < 0)
+        if (*min_max_element.first == 0.0 && *min_max_element.second == 0.0)
```

Comment:
Should we introduce a small tolerance for the check?

Reply:
I agree that, usually, one should never compare floating point number exactly. But here, also I think that we actually want to. Moving the zero-contour would just change it to one of the other values of `LocationToLevelSet`.

Regarding tracefem: If we choose to implement it like this I do not think you need to perturb the level set. If a face is aligned, you integrate over the zero contour of the cell that is chosen as intersected and not over the other cell, so it will work also for this case.